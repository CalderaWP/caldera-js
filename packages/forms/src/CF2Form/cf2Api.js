import getCf2Token from "../Http/handlers/getCf2Token";
import handleFormSubmitCf2 from "../Http/handlers/handleFormSubmitCf2";

async function cf2Api(apiRootUri, formId, axios) {

	return {
		getCf2Token,
		handleFormSubmitCf2(
			{
				entryValues,
				tokens,
			}
		) {
			return handleFormSubmitCf2({
				apiRootUri,
				formId,
				entryValues,
				tokens,
				axios
			})
		}
	}

}

export default cf2Api;
