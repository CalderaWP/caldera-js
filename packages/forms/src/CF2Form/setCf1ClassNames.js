import {
	classNameService,
	cf1ClassNames,
	FILTER_FORM_COLUMN_PREFIX,
	FILTER_FORM_COLUMN_CLASS,
	FILTER_FORM_ELEMENT_CLASS,
} from "@calderajs/components";

const IDENTIFIER = "cf1-classes";

/**
 * Sets all form classes to match Caldera Forms 1.x
 *
 * NOTE: Resets class service first.
 */
export default function setCf1ClassNames() {
	classNameService.reset();
	Object.keys(cf1ClassNames).map(filterName => {
		classNameService.addFilter(filterName, IDENTIFIER, () => {
			return cf1ClassNames[filterName];
		});
	});

	classNameService.addFilter(
		FILTER_FORM_COLUMN_PREFIX,
		IDENTIFIER,
		(prefix, columnId) => {
			return "col-sm-";
		}
	);

	/**
	 * Remove all filters
	 */
	const reset = () => {
		Object.keys(cf1ClassNames).map(filterName => {
			classNameService.removeAllFilters(filterName);
		});
	};

	return reset;
}
