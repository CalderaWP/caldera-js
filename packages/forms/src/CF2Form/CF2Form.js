import {CalderaForm} from "../CalderaForm";
import getCf2Token from "../Http/handlers/getCf2Token";
import handleFormSubmitCf2 from "../Http/handlers/handleFormSubmitCf2";
import React, {useState, useEffect, Fragment} from 'react';
import { RingLoader, PacmanLoader } from 'react-spinners';

import PropTypes from "prop-types";
import setCf1ClassNames from "./setCf1ClassNames";

/**
 * A Caldera Form designed to be processed against Caldera Forms' cf2 REST API.
 * @param apiRootUri
 * @param formConfig
 * @param axios
 * @param _tokens
 * @param onChange
 * @param onBlur
 * @param useCf1ClassNames
 * @return {*}
 * @constructor
 */
export const CF2Form = (
	{
		apiRootUri,
		formConfig,
		axios,
		_tokens,
		onChange,
		onBlur,
		useCf1ClassNames
	}
) => {

	const [formLoaded, setFormLoaded] = useState(false);
	const [tokensFetched, setTokensFetched] = useState(false);
	const [message, setMessage] = useState('');
	const [isSubmitting, setIsSubmitting] = useState(false);
	const [hideForm, setHideForm] = useState(false);
	const [tokens, setTokens] = useState(_tokens || {
		_cf_verify: '',
		_sessionPublicKey: ''
	});
	const [form, setForm] = useState(formConfig);
	const formId = form.ID;
	/**
	 * Get tokens if not passed.
	 */
	useEffect(() => {
		async function getToken() {
			getCf2Token(apiRootUri, formId, axios)
				.then(r => {
					setTokens(r);
					setTimeout(() => setTokensFetched(true), 500);
				});
		}

		if (tokens._cf_verify && tokens._sessionPublicKey) {
			setTokensFetched(true);
			return;
		}

		getToken();


	}, [tokensFetched]);

	/**
	 * Setup class service when mounting, if needed
	 */
	useEffect(() => {

		if (useCf1ClassNames) {
			setCf1ClassNames();
		}

	}, [useCf1ClassNames]);

	if (hideForm) {
		return (
			<div className={'error'}>{message}</div>

		);
	}

	const FormOrLoading = () => {
		if (isSubmitting) {
			return (
				<div><PacmanLoader/></div>
			);
		}
		return (
			<CalderaForm
				form={form}
				fields={form.fields}
				onChange={newValues => {
					onChange(onChange);
				}}
				onBlur={onBlur}
				onSubmit={(values, actions) => {
					setIsSubmitting(true);
					actions.setSubmitting(false);
					handleFormSubmitCf2({
						entryValues: values,
						tokens,
						apiRootUri,
						formId,
						axios
					}).then(r => {
						setTimeout(() => setIsSubmitting(false), 500);
						setHideForm(true);
						setMessage(r.data.message);
						actions.resetForm();
					}).catch(e => {
						setTimeout(() => setIsSubmitting(false), 500);
						console.log(e);
					});

				}}
			/>
		);

	};


	return (
		<Fragment>
			{tokensFetched ?
				<FormOrLoading/>
				: <div><RingLoader /></div>
			}
		</Fragment>
	);

};

CF2Form.propTypes = {
	...CalderaForm.propTypes,
	useCf1ClassNames: PropTypes.bool
};

const noop = () => {
};
CF2Form.defaultProps = {...CalderaForm.defaultProps, useCf1ClassNames: false};
