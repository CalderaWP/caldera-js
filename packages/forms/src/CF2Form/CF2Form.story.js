import React from 'react';
import { storiesOf } from '@storybook/react';
import { formRows } from '../CalderaForm/columns.fixtures';

import {collectFieldValues} from '@calderajs/components';
import {createFieldRule} from '../CalderaForm/state/createFieldRule';
import {emailField, textField, checkboxFieldset} from "../CalderaForm/fields.fixtures";
import {CF2Form} from "./CF2Form";
import axios from "axios";
import {render} from "react-testing-library";



const form = {
	ID: 'cf2',
	rows: [
		{
			rowId: 'r1',
			columns: [
				{
					fields: [emailField.fieldId],
					width: '1/2',
					columnId: '1a'
				},
				{
					fields: [textField.fieldId],
					width: '1/4',
					columnId: '1b'
				},
				{
					fields: [checkboxFieldset.fieldId],
					width: '1/4',
					columnId: '1c'
				}
			]
		}
	],
	fields: [
		emailField,
		textField,
		{
			...checkboxFieldset,
			value: [],
		}
	],
	conditionals: [
		{
			type: 'hide',
			rule: createFieldRule('is', emailField.fieldId, 'hide'),
			fields: [
				textField.fieldId
			]
		},

	]
};

storiesOf('CF2 Form').add('Forms', () => (
	<div>
		<p>Works with the cf2 API in Caldera Forms</p>
		<p>Not setup to submit properly here though :)</p>
		<CF2Form
			formConfig={form}
			axios={axios}
			_tokens={{
				_cf_verify: 'a',
				_sessionPublicKey: 'b'
			}}
		/>
	</div>

));

storiesOf('CF2 Form').add('With CF1 class names', () => (
	<div>
		<CF2Form
			formConfig={form}
			axios={axios}
			_tokens={{
				_cf_verify: 'a',
				_sessionPublicKey: 'b'
			}}
			useCf1ClassNames={true}
		/>
	</div>

));




storiesOf('CF2 Form').add('Resolves promise onReady', () => {
	const onReady = new Promise((resolve) => {
		console.log('loaded');
		resolve();
	});
	return(
		<CF2Form
			formConfig={formConfig}
			axios={axios}
			onReady={onReady}
			_tokens={{
				_cf_verify: 'a',
				_sessionPublicKey: 'b'
			}}
		/>
	);
});




