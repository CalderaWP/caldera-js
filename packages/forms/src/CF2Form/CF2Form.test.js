import {CF2Form} from "./CF2Form";
import React from 'react';
import renderer from 'react-test-renderer';

describe( 'CF2Form', () => {
	let axios = {
		post: jest.fn((config) => Promise.resolve({data: {}})),
		request: jest.fn((config) => Promise.resolve({data: {}}))
	};

	beforeEach(() => {
		axios = {
			post: jest.fn((config) => Promise.resolve({data: {}})),
			request: jest.fn((config) => Promise.resolve({data: {}}))
		};
	});

	const submitButton = {
		fieldId: 'fld_7908577',
		label: 'Click Me',
		fieldType: 'submit',
	};

	const formConfig = {
		ID: 'cf222',
		rows: [
			{
				rowId: 'r3',
				columns: [
					{
						fields: [submitButton.fieldId],
						width: '1',
						columnId: `r2-${submitButton.fieldId}`
					},
				]
			}
		],
		fields: [
			submitButton
		],
		conditionals: []
	};

	it( 'Matches snapshot',async (done) => {
		expect(renderer.create(<CF2Form formConfig={formConfig} axios={axios}/>)).toMatchSnapshot();
	});

	it( 'Uses injected _tokens when possible',async (done) => {
		expect(renderer.create(
			<CF2Form
				apiRootUri={'https://localhost/'}
				formConfig={formConfig}
				axios={axios}
				_tokens={{
					_cf_verify: 'a',
					_sessionPublicKey: 'b'
				}}
			/>).props
		).toBe({sivan:true})
	});
});
