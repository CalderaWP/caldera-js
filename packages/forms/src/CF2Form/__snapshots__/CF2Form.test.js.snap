// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CF2Form Matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div>
        Loading Spinner
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      Loading Spinner
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CF2Form Matches snapshot with cf2 classnames used 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div>
      <div
        class="caldera-grid"
      >
        <form
          class="caldera_forms_form"
        >
          <div
            class="sc-bdVaJa caldera-form-row row sc-bwzfXH pjOvZ"
          >
            <div
              class=" col-sm-1 sc-bdVaJa iHZvIS"
              id="r2-fld_7908577"
            >
              <div
                class="caldera-field-group caldera-field-area-submit form-group"
              >
                <label
                  class="form-label"
                  for="fld_7908577"
                >
                  Click Me
                   
                </label>
                <input
                  class="form-control form-control"
                  id="fld_7908577"
                  value="Click Me"
                />
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="caldera-grid"
    >
      <form
        class="caldera_forms_form"
      >
        <div
          class="sc-bdVaJa caldera-form-row row sc-bwzfXH pjOvZ"
        >
          <div
            class=" col-sm-1 sc-bdVaJa iHZvIS"
            id="r2-fld_7908577"
          >
            <div
              class="caldera-field-group caldera-field-area-submit form-group"
            >
              <label
                class="form-label"
                for="fld_7908577"
              >
                Click Me
                 
              </label>
              <input
                class="form-control form-control"
                id="fld_7908577"
                value="Click Me"
              />
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CF2Form Snapshot with tokens 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="caldera"
      >
        <form
          class="cf222 caldera-form"
        >
          <div
            class="sc-bdVaJa caldera-form-row caldera-row sc-bwzfXH pjOvZ"
          >
            <div
              class="caldera-column sc-bdVaJa iHZvIS"
            >
              <div
                class="caldera-field-wrapper caldera-field-wrapper-text"
              >
                <div
                  class="components-base-control"
                >
                  <div
                    class="components-base-control__field"
                  >
                    <input
                      id="fld_7908577"
                      type="submit"
                      optionValue="Click Me"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div>
      <div
        class="caldera"
      >
        <form
          class="cf222 caldera-form"
        >
          <div
            class="sc-bdVaJa caldera-form-row caldera-row sc-bwzfXH pjOvZ"
          >
            <div
              class="caldera-column sc-bdVaJa iHZvIS"
            >
              <div
                class="caldera-field-wrapper caldera-field-wrapper-text"
              >
                <div
                  class="components-base-control"
                >
                  <div
                    class="components-base-control__field"
                  >
                    <input
                      id="fld_7908577"
                      type="submit"
                      optionValue="Click Me"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="caldera"
    >
      <form
        class="cf222 caldera-form"
      >
        <div
          class="sc-bdVaJa caldera-form-row caldera-row sc-bwzfXH pjOvZ"
        >
          <div
            class="caldera-column sc-bdVaJa iHZvIS"
          >
            <div
              class="caldera-field-wrapper caldera-field-wrapper-text"
            >
              <div
                class="components-base-control"
              >
                <div
                  class="components-base-control__field"
                >
                  <input
                    id="fld_7908577"
                    type="submit"
                    optionValue="Click Me"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
